1.10.0-DEV.1432 testall throws "Load error: no active project" in Pkg and SparseArrays

This is with an isolated DEPOT_PATH during build and testing. 
Replicated outside of testall during gmake test-Pkg and test-SparseArrays

Error at ff23b3769f is new since d2f5bbd7cf

<details>
<summary>test summary</summary>

```
julia> versioninfo()
Julia Version 1.10.0-DEV.1432
Commit ff23b3769f (2023-06-04 01:27 UTC)
Platform Info:
  OS: macOS (arm64-apple-darwin21.6.0)
  CPU: 10 × Apple M1 Max
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-15.0.7 (ORCJIT, apple-m1)
  Threads: 5 on 8 virtual cores
Environment:
  JULIA_NUM_THREADS = 4
  JULIA_DEPOT_PATH = /Users/foo/applications/juliabuild

Error in testset SparseArrays/ambiguous:

Error During Test at /Users/foo/applications/julia/test/testdefs.jl:21
  Got exception outside of a @test
  LoadError: no active project

  Stacktrace:
    [1] pkgerror(msg::String)
      @ Pkg.Types ~/applications/julia/usr/share/julia/stdlib/v1.10/Pkg/src/Types.jl:69
    [2] find_project_file(env::Nothing)
      @ Pkg.Types ~/applications/julia/usr/share/julia/stdlib/v1.10/Pkg/src/Types.jl:209
    [3] Pkg.Types.EnvCache(env::Nothing)
      @ Pkg.Types ~/applications/julia/usr/share/julia/stdlib/v1.10/Pkg/src/Types.jl:345
    [4] Pkg.Types.EnvCache(env::Union{Nothing, String})
      @ Pkg.Types ~/applications/julia/usr/share/julia/stdlib/v1.10/Pkg/src/Types.jl:345 [inlined]
    [5] Context
      @ Pkg.Types ~/applications/julia/usr/share/julia/stdlib/v1.10/Pkg/src/Types.jl:395 [inlined]
    [6] add(pkgs::Vector{Pkg.Types.PackageSpec}; io::IOBuffer, kwargs::@Kwargs{})
      @ Pkg.API ~/applications/julia/usr/share/julia/stdlib/v1.10/Pkg/src/API.jl:147
    [7] add(pkgs::Vector{String}; kwargs::@Kwargs{io::IOBuffer})
      @ Pkg.API ~/applications/julia/usr/share/julia/stdlib/v1.10/Pkg/src/API.jl:144 [inlined]
    [8] add(pkg::String; kwargs::@Kwargs{io::IOBuffer})
      @ Pkg.API ~/applications/julia/usr/share/julia/stdlib/v1.10/Pkg/src/API.jl:143 [inlined]
    [9] top-level scope
      @ ~/applications/julia/usr/share/julia/stdlib/v1.10/SparseArrays/test/ambiguous.jl:17
   [10] include(mod::Module, _path::String)
      @ Base ./Base.jl:489 [inlined]
   [11] macro expansion
      @ Main ~/applications/julia/test/testdefs.jl:29 [inlined]
   [12] macro expansion
      @ Main ~/applications/julia/usr/share/julia/stdlib/v1.10/Test/src/Test.jl:1547 [inlined]
   [13] macro expansion
      @ Main ~/applications/julia/test/testdefs.jl:23 [inlined]
   [14] macro expansion
      @ Main ./timing.jl:503 [inlined]
   [15] runtests(name::String, path::String, isolate::Bool; seed::UInt128)
      @ Main ~/applications/julia/test/testdefs.jl:21
   [16] invokelatest(::Any, ::Any, ::Vararg{Any}; kwargs::@Kwargs{seed::UInt128})
      @ Base ./essentials.jl:866
   [17] (::Distributed.var"#110#112"{Distributed.CallMsg{:call_fetch}})()
      @ Distributed ~/applications/julia/usr/share/julia/stdlib/v1.10/Distributed/src/process_messages.jl:285
   [18] run_work_thunk(thunk::Distributed.var"#110#112"{Distributed.CallMsg{:call_fetch}}, print_error::Bool)
      @ Distributed ~/applications/julia/usr/share/julia/stdlib/v1.10/Distributed/src/process_messages.jl:70
   [19] (::Distributed.var"#109#111"{Distributed.CallMsg{:call_fetch}, Distributed.MsgHeader, Sockets.TCPSocket})()
      @ Distributed ~/applications/julia/usr/share/julia/stdlib/v1.10/Distributed/src/process_messages.jl:285
  in expression starting at /Users/foo/applications/julia/usr/share/julia/stdlib/v1.10/SparseArrays/test/ambiguous.jl:13


Error in testset Pkg:

Error During Test at /Users/foo/applications/julia/test/testdefs.jl:21
  Got exception outside of a @test
  LoadError: no active project

  Stacktrace:
    [1] pkgerror(msg::String)
      @ Pkg.Types ~/applications/julia/usr/share/julia/stdlib/v1.10/Pkg/src/Types.jl:69
    [2] find_project_file(env::Nothing)
      @ Pkg.Types ~/applications/julia/usr/share/julia/stdlib/v1.10/Pkg/src/Types.jl:209
    [3] Pkg.Types.EnvCache(env::Nothing)
      @ Pkg.Types ~/applications/julia/usr/share/julia/stdlib/v1.10/Pkg/src/Types.jl:345
    [4] Pkg.Types.EnvCache(env::Union{Nothing, String})
      @ Pkg.Types ~/applications/julia/usr/share/julia/stdlib/v1.10/Pkg/src/Types.jl:345 [inlined]
    [5] Context
      @ Pkg.Types ~/applications/julia/usr/share/julia/stdlib/v1.10/Pkg/src/Types.jl:395 [inlined]
    [6] add(pkgs::Vector{Pkg.Types.PackageSpec}; io::IOBuffer, kwargs::@Kwargs{})
      @ Pkg.API ~/applications/julia/usr/share/julia/stdlib/v1.10/Pkg/src/API.jl:147
    [7] add(pkgs::Vector{String}; kwargs::@Kwargs{io::IOBuffer})
      @ Pkg.API ~/applications/julia/usr/share/julia/stdlib/v1.10/Pkg/src/API.jl:144 [inlined]
    [8] add(pkg::String; kwargs::@Kwargs{io::IOBuffer})
      @ Pkg.API ~/applications/julia/usr/share/julia/stdlib/v1.10/Pkg/src/API.jl:143 [inlined]
    [9] top-level scope
      @ ~/applications/julia/usr/share/julia/stdlib/v1.10/Pkg/test/runtests.jl:18
   [10] include(mod::Module, _path::String)
      @ Base ./Base.jl:489 [inlined]
   [11] macro expansion
      @ Main ~/applications/julia/test/testdefs.jl:29 [inlined]
   [12] macro expansion
      @ Main ~/applications/julia/usr/share/julia/stdlib/v1.10/Test/src/Test.jl:1547 [inlined]
   [13] macro expansion
      @ Main ~/applications/julia/test/testdefs.jl:23 [inlined]
   [14] macro expansion
      @ Main ./timing.jl:503 [inlined]
   [15] runtests(name::String, path::String, isolate::Bool; seed::UInt128)
      @ Main ~/applications/julia/test/testdefs.jl:21
   [16] invokelatest(::Any, ::Any, ::Vararg{Any}; kwargs::@Kwargs{seed::UInt128})
      @ Base ./essentials.jl:866
   [17] (::Distributed.var"#110#112"{Distributed.CallMsg{:call_fetch}})()
      @ Distributed ~/applications/julia/usr/share/julia/stdlib/v1.10/Distributed/src/process_messages.jl:285
   [18] run_work_thunk(thunk::Distributed.var"#110#112"{Distributed.CallMsg{:call_fetch}}, print_error::Bool)
      @ Distributed ~/applications/julia/usr/share/julia/stdlib/v1.10/Distributed/src/process_messages.jl:70
   [19] (::Distributed.var"#109#111"{Distributed.CallMsg{:call_fetch}, Distributed.MsgHeader, Sockets.TCPSocket})()
      @ Distributed ~/applications/julia/usr/share/julia/stdlib/v1.10/Distributed/src/process_messages.jl:285
  in expression starting at /Users/foo/applications/julia/usr/share/julia/stdlib/v1.10/Pkg/test/runtests.jl:3
ERROR: LoadError: Test run finished with errors
in expression starting at /Users/foo/applications/julia/test/runtests.jl:94
gmake[1]: *** [Makefile:30: all] Error 1
gmake: *** [Makefile:611: testall] Error 2

```
  
</details>

Still present on 1.10.0-beta1
Details of build environment and test

<details>
<summary>build details</summary>

```
% git branch                                                                                                                                                                                        0:32
  master
* release-1.10

% git log
commit 6616549950e6e91eb7ab06708fa064b9d75c6282 (HEAD -> release-1.10, tag: v1.10.0-beta1, origin/release-1.10)
Author: Kristoffer Carlsson <kristoffer.carlsson@chalmers.se>
Date:   Tue Jul 25 13:43:26 2023 -0400

    release-1.10: set VERSON to 1.10.0-beta1 (#50662)

% export JULIA_DEPOT_PATH=/Users/foo/applications/juliabuild

% gmake

% ./julia

julia> versioninfo(verbose=true)
Julia Version 1.10.0-beta1
Commit 6616549950 (2023-07-25 17:43 UTC)
Platform Info:
  OS: macOS (arm64-apple-darwin21.6.0)
  uname: Darwin 21.6.0 Darwin Kernel Version 21.6.0: Thu Jul  6 22:20:00 PDT 2023; root:xnu-8020.240.18.702.13~1/RELEASE_ARM64_T6000 arm64 arm
  CPU: Apple M1 Max:
                 speed         user         nice          sys         idle          irq
       #1-10  2400 MHz     431434 s          0 s     262373 s   11560660 s          0 s
  Memory: 64.0 GB (43428.046875 MB free)
  Uptime: 132720.0 sec
  Load Avg:  1.04248046875  1.96728515625  3.13720703125
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-15.0.7 (ORCJIT, apple-m1)
  Threads: 5 on 8 virtual cores
Environment:
  JULIA_NUM_THREADS = 4
  JULIA_DEPOT_PATH = /Users/foo/applications/juliabuild
  HOME = /Users/foo
  PATH = /Users/foo/.pyenv/bin:/opt/homebrew/opt/grep/libexec/gnubin:/Users/foo/bin:/Users/foo/.local/bin:/opt/homebrew/bin:/Users/foo/Library/Python/3.9/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/bin:/Applications/MacPorts/EmacsMac.app/Contents/MacOS:/opt/local/bin:/opt/local/sbin:/opt/homebrew/bin:/opt/homebrew/sbin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/TeX/texbin:/opt/X11/bin:/Library/Apple/usr/bin:/Users/foo/perl5/perlbrew/bin:/Users/foo/perl5/perlbrew/perls/perl-5.35.5/bin:/Users/foo/.local/bin:/opt/homebrew/opt/fzf/bin:/Users/foo/.config/emacs/bin
  TERM = xterm-256color
  XPC_FLAGS = 0x0
  PERLBREW_HOME = /Users/foo/.perlbrew
  PERLBREW_MANPATH = /Users/foo/perl5/perlbrew/perls/perl-5.35.5/man
  PERLBREW_PATH = /Users/foo/perl5/perlbrew/bin:/Users/foo/perl5/perlbrew/perls/perl-5.35.5/bin
  MANPATH = /opt/local/share/man:/opt/homebrew/share/man:/usr/share/man:/usr/local/share/man:/Library/TeX/Distributions/.DefaultTeX/Contents/Man:/opt/X11/share/man:/Users/foo/perl5/perlbrew/perls/perl-5.35.5/man:/Library/Apple/usr/share/man:/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/share/man:/Applications/Xcode.app/Contents/Developer/usr/share/man:/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/share/man:
  HOMEBREW_PREFIX = /opt/homebrew
  HOMEBREW_CELLAR = /opt/homebrew/Cellar
  HOMEBREW_REPOSITORY = /opt/homebrew
  INFOPATH = /opt/homebrew/share/info:

% gmake docs
```
  
</details>

<details>
<summary>test summary</summary>

```
% gmake testall

Summary:                                     |     Pass  Error  Broken     Total      Time
  Overall                                         | 40629211      2    2135  40631348  10m41.5s
    LinearAlgebra/addmul                          |     2943           1719      4662   1m15.8s
    LinearAlgebra/special                         |     3675                     3675   1m51.0s
    LinearAlgebra/qr                              |     4785                     4785   1m03.5s
    LinearAlgebra/symmetric                       |     3044                     3044   2m20.5s
    LinearAlgebra/dense                           |     8571                     8571   2m45.2s
    LinearAlgebra/blas                            |     1630                     1630     27.1s
    LinearAlgebra/diagonal                        |     2957                     2957   2m47.4s
    LinearAlgebra/uniformscaling                  |      446                      446     15.9s
    LinearAlgebra/cholesky                        |     2548                     2548   1m16.7s
    LinearAlgebra/matmul                          |     1594              3      1597   3m15.3s
    LinearAlgebra/hessenberg                      |      667                      667     36.5s
    LinearAlgebra/lu                              |     1380                     1380   1m06.7s
    LinearAlgebra/svd                             |      606                      606     35.4s
    LinearAlgebra/structuredbroadcast             |      672                      672     53.4s
    LinearAlgebra/lapack                          |      839                      839     17.4s
    LinearAlgebra/tridiag                         |     1637                     1637     30.6s
    LinearAlgebra/eigen                           |      951                      951     38.3s
    LinearAlgebra/lq                              |     2898                     2898     20.4s
    LinearAlgebra/givens                          |     1991                     1991      3.6s
    LinearAlgebra/factorization                   |       85             16       101      1.4s
    LinearAlgebra/adjtrans                        |      485                      485     14.7s
    LinearAlgebra/ldlt                            |        8                        8      0.4s
    LinearAlgebra/pinv                            |      500                      500      5.0s
    LinearAlgebra/symmetriceigen                  |       16                       16      1.4s
    compiler/datastructures                       |      283                      283      1.0s
    LinearAlgebra/abstractq                       |      120                      120      3.3s
    ambiguous                                     |      119              2       121      1.9s
    compiler/validation                           |       27                       27      0.2s
    compiler/effects                              |      679              1       680      1.3s
    LinearAlgebra/schur                           |      500                      500     14.8s
    compiler/irpasses                             |      263              4       267      2.5s
    compiler/ssair                                |      140                      140      3.8s
    compiler/contextual                           |       11                       11      1.4s
    compiler/codegen                              |      188                      188      6.5s
    LinearAlgebra/generic                         |      724              1       725     23.6s
    compiler/invalidation                         |       44                       44      3.8s
    compiler/inline                               |      305              3       308      6.6s
    compiler/inference                            |     1418              6      1424     11.4s
    compiler/EscapeAnalysis/interprocedural       |       32              4        36      5.5s
    strings/basic                                 |    93504                    93504      5.1s
    strings/search                                |      876                      876      2.3s
    compiler/EscapeAnalysis/local                 |      346             21       367      8.8s
    LinearAlgebra/bunchkaufman                    |     5689                     5689     26.9s
    unicode/utf8                                  |       19                       19      0.1s
    strings/io                                    |    12765                    12765      1.4s
    worlds                                        |      103                      103      0.8s
    strings/types                                 |  2302691                  2302691      1.7s
    keywordargs                                   |      155                      155      0.8s
    strings/util                                  |     1154                     1154      4.5s
    compiler/AbstractInterpreter                  |       30                       30     15.3s
    char                                          |    10855                    10855      1.5s
    triplequote                                   |       29                       29      0.0s
    subtype                                       |   317367             50    317417      4.9s
    intrinsics                                    |      371                      371      1.3s
    atomics                                       |     3448                     3448      7.4s
    iobuffer                                      |      210                      210      0.7s
    staged                                        |       69              1        70      1.2s
    hashing                                       |    14445                    14445      8.5s
    dict                                          |   144466                   144466     12.0s
    tuple                                         |      683                      683      3.8s
    LinearAlgebra/bidiag                          |     7326                     7326   4m38.0s
    offsetarray                                   |      513              3       516     19.3s
    reduce                                        |     8602                     8602      9.1s
    numbers                                       |  1085252              1   1085253     31.5s
    simdloop                                      |      240                      240      0.8s
    core                                          |  8446056              3   8446059     35.8s
    vecelement                                    |      678                      678      3.2s
    intfuncs                                      |   228578                   228578     14.1s
    copy                                          |      556                      556      1.4s
    rational                                      |    98706              2     98708     12.6s
    fastmath                                      |      965                      965      5.7s
    functional                                    |      101                      101      2.8s
    math                                          |  1915663                  1915663     17.5s
    reducedim                                     |     1225              6      1231     43.0s
    operators                                     |    13072                    13072      4.0s
    path                                          |      377             12       389      0.6s
    ordering                                      |       37                       37      2.7s
    parse                                         |    16099                    16099      1.5s
    arrayops                                      |     2137              5      2142     55.0s
    gmp                                           |     2459                     2459      2.7s
    abstractarray                                 |    55349             91     55440     48.3s
    backtrace                                     |       40              1        41      0.9s
    exceptions                                    |       70                       70      1.0s
    file                                          |     1076                     1076      4.0s
    iterators                                     |    86621                    86621     39.6s
    version                                       |     2453                     2453      0.5s
    namedtuple                                    |      242                      242      2.2s
    read                                          |     3812                     3812     16.3s
    mpfr                                          |     1137              1      1138      7.6s
    subarray                                      |   459022                   459022   1m57.2s
    floatapprox                                   |       49                       49      0.5s
    complex                                       |     8496              2      8498      4.9s
    reflection                                    |      451                      451      2.8s
    regex                                         |      155                      155      1.2s
    combinatorics                                 |      178                      178      2.9s
    sysinfo                                       |        7                        7      0.1s
    loading                                       |   161314                   161314     45.1s
    env                                           |      169                      169      0.5s
    float16                                       |   762093                   762093      4.7s
    mod2pi                                        |       80                       80      1.3s
    rounding                                      |   150016                   150016      2.6s
    euler                                         |       12                       12      0.5s
    client                                        |        5                        5      0.8s
    errorshow                                     |      278                      278      3.5s
    show                                          |   128994              8    129002     15.1s
    goto                                          |       19                       19      0.0s
    llvmcall                                      |       21                       21      0.3s
    llvmcall2                                     |       21                       21      0.1s
    ryu                                           |    31217                    31217      0.9s
    sets                                          |     3779                     3779     12.1s
    some                                          |       72                       72      0.5s
    meta                                          |       75                       75      0.7s
    stacktraces                                   |       72              6        78      1.4s
    docs                                          |      239                      239      2.5s
    binaryplatforms                               |      346                      346      2.2s
    ranges                                        | 12438702              3  12438705     25.8s
    enums                                         |      102                      102      1.5s
    atexit                                        |       68                       68      6.6s
    int                                           |   736488             40    736528      5.9s
    interpreter                                   |        3                        3      0.6s
    checked                                       |     1239                     1239      4.6s
    misc                                          |  1282255                  1282255     22.2s
    bitset                                        |      206                      206      1.0s
    sorting                                       |    41519              9     41528   1m29.3s
    error                                         |       84                       84      1.2s
    broadcast                                     |      528                      528   1m01.9s
    osutils                                       |       43                       43      0.0s
    boundscheck                                   |                              None      4.7s
    iostream                                      |       50                       50      0.6s
    cartesian                                     |      351              3       354      3.4s
    secretbuffer                                  |       30                       30      0.8s
    specificity                                   |      182                      182      0.1s
    syntax                                        |     1699              1      1700      4.3s
    corelogging                                   |      235                      235      2.2s
    reinterpretarray                              |      431                      431      9.4s
    channels                                      |      260                      260     15.2s
    smallarrayshrink                              |       36                       36      0.2s
    opaque_closure                                |       82              1        83      0.4s
    filesystem                                    |        6                        6      0.1s
    asyncmap                                      |      307                      307      5.5s
    bitarray                                      |   939768              8    939776   2m23.1s
    SparseArrays/allowscalar                      |        9                        9      0.5s
    missing                                       |      580              1       581      9.8s
    SparseArrays/ambiguous                        |               1                 1      3.2s
    SparseArrays/fixed                            |       77                       77      3.9s
    spawn                                         |      252              4       256   1m59.8s
    SparseArrays/cholmod                          |      668                      668     19.5s
    download                                      |                              None     31.1s
    SparseArrays/linalg_solvers                   |      364                      364     10.6s
    SparseArrays/issues                           |      350                      350     31.1s
    SparseArrays/linalg                           |     3472                     3472     46.2s
    SparseArrays/sparsematrix_ops                 |      675                      675     29.8s
    SparseArrays/spqr                             |      168                      168     16.0s
    LibGit2/online                                |       10                       10      2.5s
    SparseArrays/umfpack                          |      349                      349     23.3s
    SparseArrays/higherorderfns                   |     7172              1      7173   1m21.3s
    Dates/accessors                               |  7723858                  7723858      4.5s
    Dates/query                                   |     1004                     1004      0.7s
    Dates/adjusters                               |     3149                     3149      2.7s
    LinearAlgebra/triangular                      |    38457                    38457   8m45.6s
    Dates/rounding                                |      315                      315      1.2s
    Dates/types                                   |      232                      232      0.9s
    Dates/ranges                                  |   350640                   350640      8.2s
    Dates/periods                                 |      953                      953     10.0s
    Dates/conversions                             |      161                      161      0.4s
    Dates/io                                      |      361                      361      5.7s
    Dates/arithmetic                              |      386                      386      2.9s
    Base64                                        |     2031                     2031      1.1s
    CRC32c                                        |      664                      664      0.3s
    ArgTools                                      |      180                      180      3.2s
    CompilerSupportLibraries_jll                  |        4                        4      0.8s
    Artifacts                                     |     1452                     1452      2.7s
    SparseArrays/sparsematrix_constructors_indexing |     2112                     2112   1m14.8s
    Future                                        |                              None      0.0s
    GMP_jll                                       |        1                        1      0.0s
    DelimitedFiles                                |       89                       89      2.0s
    LLD_jll                                       |        2                        2      0.4s
    LLVMLibUnwind_jll                             |        7                        7      0.1s
    LazyArtifacts                                 |        4                        4      1.5s
    LibCURL                                       |        6                        6      1.1s
    LibCURL_jll                                   |        1                        1      0.0s
    LibGit2_jll                                   |        2                        2      0.5s
    LibSSH2_jll                                   |                              None      0.0s
    LibUV_jll                                     |        1                        1      0.2s
    LibUnwind_jll                                 |                              None      0.1s
    cmdlineargs                                   |      312              6       318   2m25.8s
    Libdl                                         |       99              1       100      0.8s
    InteractiveUtils                              |      300                      300      5.5s
    MPFR_jll                                      |        1                        1      0.1s
    MbedTLS_jll                                   |        1                        1      0.0s
    Logging                                       |       40                       40      0.6s
    MozillaCACerts_jll                            |        1                        1      0.0s
    NetworkOptions                                |     3518                     3518      0.6s
    OpenBLAS_jll                                  |        1                        1      0.0s
    OpenLibm_jll                                  |        1                        1      0.1s
    PCRE2_jll                                     |        2                        2      0.1s
    Markdown                                      |      262                      262      2.4s
    Pkg                                           |               1                 1      4.6s
    Mmap                                          |      140                      140     14.3s
    Printf                                        |     1316                     1316     12.4s
    REPL                                          |     1629              3      1632     11.7s
    floatfuncs                                    |      232                      232   2m40.9s
    Profile                                       |      116                      116     24.3s
    LibGit2/libgit2                               |      758                      758   1m04.3s
    Serialization                                 |      131              1       132      2.8s
    SuiteSparse                                   |                              None      1.6s
    SuiteSparse_jll                               |        1                        1      0.0s
    Sockets                                       |      172                      172      4.8s
    Random                                        |   204875                   204875     25.3s
    TOML                                          |      623             45       668      8.4s
    UUIDs                                         |     1029                     1029      0.2s
    SparseArrays/sparsevector                     |    11157              4     11161   1m58.1s
    Zlib_jll                                      |        1                        1      0.1s
    dSFMT_jll                                     |        1                        1      0.1s
    Unicode                                       |      795                      795      2.1s
    libblastrampoline_jll                         |        1                        1      0.0s
    nghttp2_jll                                   |        1                        1      0.0s
    p7zip_jll                                     |        1                        1      0.0s
    libLLVM_jll                                   |        1                        1      0.3s
    Tar                                           |     3225             11      3236     12.8s
    Test                                          |      494             17       511     19.3s
    Statistics                                    |      810                      810     27.4s
    Downloads                                     |      240                      240   1m08.7s
    SHA                                           |      107                      107     44.5s
    FileWatching                                  |      562                      562   1m44.9s
    ccall                                         |   165137              1    165138      9.3s
    precompile                                    |      243                      243     29.4s
    SharedArrays                                  |      114                      114     10.8s
    threads                                       |       34              3        37   1m43.0s
    Distributed                                   |       15                       15   1m57.6s
    gc                                            |       24                       24     13.4s
    stress                                        |      118                      118      8.1s
    FAILURE

The global RNG seed was 0x6b3fdd801c076c49fabdbeaf663a34e.

Error in testset SparseArrays/ambiguous:
Error During Test at /Users/foo/applications/juliadev/test/testdefs.jl:21
  Got exception outside of a @test
  LoadError: no active project
  Stacktrace:
    [1] pkgerror(msg::String)
      @ Pkg.Types ~/applications/juliadev/usr/share/julia/stdlib/v1.10/Pkg/src/Types.jl:69
    [2] find_project_file(env::Nothing)
      @ Pkg.Types ~/applications/juliadev/usr/share/julia/stdlib/v1.10/Pkg/src/Types.jl:209
    [3] Pkg.Types.EnvCache(env::Nothing) (repeats 2 times)
      @ Pkg.Types ~/applications/juliadev/usr/share/julia/stdlib/v1.10/Pkg/src/Types.jl:345
    [4] Context
      @ Pkg.API ~/applications/juliadev/usr/share/julia/stdlib/v1.10/Pkg/src/Types.jl:395 [inlined]
    [5] add(pkgs::Vector{Pkg.Types.PackageSpec}; io::IOBuffer, kwargs::@Kwargs{})
      @ Pkg.API ~/applications/juliadev/usr/share/julia/stdlib/v1.10/Pkg/src/API.jl:150
    [6] add
      @ Pkg.API ~/applications/juliadev/usr/share/julia/stdlib/v1.10/Pkg/src/API.jl:148 [inlined]
    [7] add
      @ Pkg.API ~/applications/juliadev/usr/share/julia/stdlib/v1.10/Pkg/src/API.jl:147 [inlined]
    [8] #add#24
      @ Pkg.API ~/applications/juliadev/usr/share/julia/stdlib/v1.10/Pkg/src/API.jl:146 [inlined]
    [9] top-level scope
      @ ~/applications/juliadev/usr/share/julia/stdlib/v1.10/SparseArrays/test/ambiguous.jl:17
   [10] include
      @ Main ./Base.jl:489 [inlined]
   [11] macro expansion
      @ Main ~/applications/juliadev/test/testdefs.jl:29 [inlined]
   [12] macro expansion
      @ Main ~/applications/juliadev/usr/share/julia/stdlib/v1.10/Test/src/Test.jl:1577 [inlined]
   [13] macro expansion
      @ Main ~/applications/juliadev/test/testdefs.jl:23 [inlined]
   [14] macro expansion
      @ Main ./timing.jl:503 [inlined]
   [15] runtests(name::String, path::String, isolate::Bool; seed::UInt128)
      @ Main ~/applications/juliadev/test/testdefs.jl:21
   [16] runtests
      @ Main ~/applications/juliadev/test/testdefs.jl:5 [inlined]
   [17] invokelatest(::Any, ::Any, ::Vararg{Any}; kwargs::@Kwargs{seed::UInt128})
      @ Base ./essentials.jl:889
   [18] (::Distributed.var"#110#112"{Distributed.CallMsg{:call_fetch}})()
      @ Distributed ~/applications/juliadev/usr/share/julia/stdlib/v1.10/Distributed/src/process_messages.jl:285
   [19] run_work_thunk(thunk::Distributed.var"#110#112"{Distributed.CallMsg{:call_fetch}}, print_error::Bool)
      @ Distributed ~/applications/juliadev/usr/share/julia/stdlib/v1.10/Distributed/src/process_messages.jl:70
   [20] (::Distributed.var"#109#111"{Distributed.CallMsg{:call_fetch}, Distributed.MsgHeader, Sockets.TCPSocket})()
      @ Distributed ~/applications/juliadev/usr/share/julia/stdlib/v1.10/Distributed/src/process_messages.jl:285
  in expression starting at /Users/foo/applications/juliadev/usr/share/julia/stdlib/v1.10/SparseArrays/test/ambiguous.jl:13
Error in testset Pkg:
Error During Test at /Users/foo/applications/juliadev/test/testdefs.jl:21
  Got exception outside of a @test
  LoadError: no active project
  Stacktrace:
    [1] pkgerror(msg::String)
      @ Pkg.Types ~/applications/juliadev/usr/share/julia/stdlib/v1.10/Pkg/src/Types.jl:69
    [2] find_project_file(env::Nothing)
      @ Pkg.Types ~/applications/juliadev/usr/share/julia/stdlib/v1.10/Pkg/src/Types.jl:209
    [3] Pkg.Types.EnvCache(env::Nothing) (repeats 2 times)
      @ Pkg.Types ~/applications/juliadev/usr/share/julia/stdlib/v1.10/Pkg/src/Types.jl:345
    [4] Context
      @ Pkg.API ~/applications/juliadev/usr/share/julia/stdlib/v1.10/Pkg/src/Types.jl:395 [inlined]
    [5] add(pkgs::Vector{Pkg.Types.PackageSpec}; io::IOBuffer, kwargs::@Kwargs{})
      @ Pkg.API ~/applications/juliadev/usr/share/julia/stdlib/v1.10/Pkg/src/API.jl:150
    [6] add
      @ Pkg.API ~/applications/juliadev/usr/share/julia/stdlib/v1.10/Pkg/src/API.jl:148 [inlined]
    [7] add
      @ Pkg.API ~/applications/juliadev/usr/share/julia/stdlib/v1.10/Pkg/src/API.jl:147 [inlined]
    [8] #add#24
      @ Pkg.API ~/applications/juliadev/usr/share/julia/stdlib/v1.10/Pkg/src/API.jl:146 [inlined]
    [9] top-level scope
      @ ~/applications/juliadev/usr/share/julia/stdlib/v1.10/Pkg/test/runtests.jl:18
   [10] include
      @ Main ./Base.jl:489 [inlined]
   [11] macro expansion
      @ Main ~/applications/juliadev/test/testdefs.jl:29 [inlined]
   [12] macro expansion
      @ Main ~/applications/juliadev/usr/share/julia/stdlib/v1.10/Test/src/Test.jl:1577 [inlined]
   [13] macro expansion
      @ Main ~/applications/juliadev/test/testdefs.jl:23 [inlined]
   [14] macro expansion
      @ Main ./timing.jl:503 [inlined]
   [15] runtests(name::String, path::String, isolate::Bool; seed::UInt128)
      @ Main ~/applications/juliadev/test/testdefs.jl:21
   [16] runtests
      @ Main ~/applications/juliadev/test/testdefs.jl:5 [inlined]
   [17] invokelatest(::Any, ::Any, ::Vararg{Any}; kwargs::@Kwargs{seed::UInt128})
      @ Base ./essentials.jl:889
   [18] (::Distributed.var"#110#112"{Distributed.CallMsg{:call_fetch}})()
      @ Distributed ~/applications/juliadev/usr/share/julia/stdlib/v1.10/Distributed/src/process_messages.jl:285
   [19] run_work_thunk(thunk::Distributed.var"#110#112"{Distributed.CallMsg{:call_fetch}}, print_error::Bool)
      @ Distributed ~/applications/juliadev/usr/share/julia/stdlib/v1.10/Distributed/src/process_messages.jl:70
   [20] (::Distributed.var"#109#111"{Distributed.CallMsg{:call_fetch}, Distributed.MsgHeader, Sockets.TCPSocket})()
      @ Distributed ~/applications/juliadev/usr/share/julia/stdlib/v1.10/Distributed/src/process_messages.jl:285
  in expression starting at /Users/foo/applications/juliadev/usr/share/julia/stdlib/v1.10/Pkg/test/runtests.jl:3
ERROR: LoadError: Test run finished with errors
in expression starting at /Users/foo/applications/juliadev/test/runtests.jl:95
gmake[1]: *** [Makefile:30: all] Error 1
gmake: *** [Makefile:623: testall] Error 2
```
  
</details>

--- a/Make.inc
+++ b/Make.inc
@@ -599,10 +599,13 @@ CPP_STDOUT := $(CPP) -P
 # file extensions
 ifeq ($(OS), WINNT)
   SHLIB_EXT := dll
+  PATHSEP := ;
 else ifeq ($(OS), Darwin)
   SHLIB_EXT := dylib
+  PATHSEP := :
 else
   SHLIB_EXT := so
+  PATHSEP := :
 endif
 
 ifeq ($(OS),WINNT)

--- a/base/util.jl
+++ b/base/util.jl
@@ -691,7 +691,7 @@ function runtests(tests = ["all"]; ncores::Int = ceil(Int, Sys.CPU_THREADS / 2),
     ENV2["JULIA_CPU_THREADS"] = "$ncores"
     pathsep = Sys.iswindows() ? ";" : ":"
     ENV2["JULIA_DEPOT_PATH"] = string(mktempdir(; cleanup = true), pathsep) # make sure the default depots can be loaded
-    delete!(ENV2, "JULIA_LOAD_PATH")
+    ENV2["JULIA_LOAD_PATH"] = string("@", pathsep, "@stdlib")
     delete!(ENV2, "JULIA_PROJECT")
     try
         run(setenv(`$(julia_cmd()) $(joinpath(Sys.BINDIR,

--- a/test/Makefile
+++ b/test/Makefile
@@ -7,7 +7,7 @@ STDLIBDIR := $(build_datarootdir)/julia/stdlib/$(VERSDIR)
 # TODO: this Makefile ignores BUILDDIR, except for computing JULIA_EXECUTABLE
 
 export JULIA_DEPOT_PATH := $(build_prefix)/share/julia
-export JULIA_LOAD_PATH := @stdlib
+export JULIA_LOAD_PATH := @$(PATHSEP)@stdlib
 unexport JULIA_PROJECT :=
 unexport JULIA_BINDIR :=
 

--- a/test/loading.jl
+++ b/test/loading.jl
@@ -1277,6 +1277,9 @@ end
 
 @testset "relocatable upgrades #51989" begin
     mktempdir() do depot
+        project_path = joinpath(depot, "project")
+        mkpath(project_path)
+
         # Create fake `Foo.jl` package with two files:
         foo_path = joinpath(depot, "dev", "Foo")
         mkpath(joinpath(foo_path, "src"))
@@ -1300,9 +1303,8 @@ end
 
         # In our depot, `dev` and then `precompile` this `Foo` package.
         @test success(addenv(
-            `$(Base.julia_cmd()) --startup-file=no -e 'import Pkg; Pkg.develop("Foo"); Pkg.precompile(); exit(0)'`,
-            "JULIA_DEPOT_PATH" => depot,
-        ))
+            `$(Base.julia_cmd()) --project=$project_path --startup-file=no -e 'import Pkg; Pkg.develop("Foo"); Pkg.precompile(); exit(0)'`,
+            "JULIA_DEPOT_PATH" => depot))
 
         # Get the size of the generated `.ji` file so that we can ensure that it gets altered
         foo_compiled_path = joinpath(depot, "compiled", "v$(VERSION.major).$(VERSION.minor)", "Foo")
@@ -1321,7 +1323,7 @@ end
 
         # Try to load `Foo`; this should trigger recompilation, not an error!
         @test success(addenv(
-            `$(Base.julia_cmd()) --startup-file=no -e 'using Foo; exit(0)'`,
+            `$(Base.julia_cmd()) --project=$project_path --startup-file=no -e 'using Foo; exit(0)'`,
             "JULIA_DEPOT_PATH" => depot,
         ))
 

